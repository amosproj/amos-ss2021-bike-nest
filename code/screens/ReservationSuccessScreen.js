import React, { useState } from 'react';
import { ImageBackground, StyleSheet, Text, View, Image, TouchableOpacity,  Linking } from 'react-native';
import { Dimensions } from "react-native";
import bike from '../assets/bike.png'; 
import global from '../components/GlobalVars';
import { mainStyles } from "../styles/MainStyles";
import BikeNest_NavigationFooter from '../components/BikeNest_NavigationFooter';
import colors from '../styles/Colors';
import BikeNest_Button, { ButtonStyle } from '../components/BikeNest_Button';
import {BookingService} from "../services/BookingService";


var width = Dimensions.get('window').width; //full width
var height = Dimensions.get('window').height; //full height

export default function ReservationSuccessScreen({ route, navigation }) {
  let bookingService = new BookingService();

  let bikenest = route.params.state;
  let bikenestName = route.params.name;

  let nestAdresse = bikenestName;

  const [spotNum, setSpotNumber] = useState(0);

  let tryGETBooking = () => {
    console.log('start pulling reservation info');

    bookingService.getAllReservations().then(response => {
      if(response.success){
        // alert(JSON.stringify(response.reservations));
        // console.log(response.reservations);
        
        //TODO set spot number where reservation not expired
        setSpotNumber(response.reservations[0].bikespotId);
      }else{
        console.log(response.error);
      }
    });
}
tryGETBooking();

  return (
    <View style={mainStyles.container}>
      <View style={styles.historyContainer}>
        <View style={{alignSelf:'center'}}>
            <Text style={mainStyles.h2}>
                Reservierung <Image source={require('../assets/payment/info.png')}/>
            </Text>
        </View>
        <TouchableOpacity onPress={() => navigation.navigate("Lock") }
          style={styles.heightBike, {
            backgroundColor: '#FFF',
            height: 230,
            width: 370,
            margin: 10,
            position: 'relative'
          }}>
          <ImageBackground
            source={bike}
            style={{
              height: 230,
              width: 370,
              position: 'absolute',
            }}
          />
          <View style={{ flex: 1, alignItems: 'flex-start', justifyContent: 'flex-start', padding: 30}}>
            <Text style={{fontSize: 16, color: '#000000'}}>Vielen Dank für deine Reservierung! </Text>
            <Text style={{fontSize: 16, color: '#000000'}}>Begib dich zu folgendem BIKE NEST um dein Fahrrad abzustellen: {"\n"}</Text>
            <Text style={{fontWeight: 'bold', fontSize: 16, color: '#000000'}}>BIKE NEST {"\n"}{nestAdresse} {"\n"}Spot Nummer {spotNum}</Text>
          </View>
        </TouchableOpacity>
        <View style={styles.reserved}>
            <Image source={require('../assets/payment/clock.png')} style={{margin: 10}} />
            <Text style={mainStyles.stdText, {color: colors.UI_Light_2}}>
                Reserviert für 30min
            </Text>
        </View>
        <View style={{alignSelf:'center'}}>
            <Text style={mainStyles.stdText}>Bitte begib dich zu dem oben genannten BIKE NEST. Du kannst dich Navigieren lassen oder es direkt aufschließen, falls du schon hier bist.</Text>
        </View>
        <View style={{justifyContent: 'center', alignItems: 'center'}}>
            <BikeNest_Button overrideButtonColor={colors.UI_Light_4} 
                             overrideTextColor={colors.UI_BaseGrey_0} 
                             type={ButtonStyle.big} 
                             text="Navigation"  
                             onPress={() => { Linking.openURL('https://www.google.com/maps/dir//' + bikenest.coordinate.latitude + ',' + bikenest.coordinate.longitude) }}/>
            <BikeNest_Button overrideButtonColor={colors.UI_Light_4} 
                             overrideTextColor={colors.UI_BaseGrey_0} 
                             type={ButtonStyle.big} 
                             text="Ich bin schon hier" 
                             onPress={() => navigation.navigate("Lock")} />
        </View>
      </View>
      <BikeNest_NavigationFooter></BikeNest_NavigationFooter>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // backgroundColor: 'red',
    alignItems: 'center',
    justifyContent: 'center',
  },
  historyContainer: {
    flex: 1,
    padding: 15,
    marginTop: 30,
    justifyContent: 'space-evenly',
    position: 'relative',
  },
  reserved: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center', 
    },
});