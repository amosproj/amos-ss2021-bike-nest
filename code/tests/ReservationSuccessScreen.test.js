import React from 'react';
import renderer from 'react-test-renderer';
import ReservationSuccessScreen from '../screens/ReservationSuccessScreen';

it('renders correctly', () => {
    const params = { state: jest.fn(), name: jest.fn() };
    const tree = renderer
        .create(<ReservationSuccessScreen route={{params}}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});