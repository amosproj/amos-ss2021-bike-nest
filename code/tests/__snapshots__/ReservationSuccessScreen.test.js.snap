// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-evenly",
        "marginTop": 30,
        "padding": 15,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 27,
            "fontWeight": "400",
            "letterSpacing": 2,
          }
        }
      >
        Reservierung 
        <Image
          source={1}
        />
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFF",
          "height": 230,
          "margin": 10,
          "opacity": 1,
          "position": "relative",
          "width": 370,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": 230,
            "position": "absolute",
            "width": 370,
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 230,
                "width": 370,
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "flex-start",
            "padding": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
            }
          }
        >
          Vielen Dank für deine Reservierung! 
        </Text>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
            }
          }
        >
          Begib dich zu folgendem BIKE NEST um dein Fahrrad abzustellen: 
          

        </Text>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          BIKE NEST 
          

           
          

          Spot Nummer 
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "margin": 10,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
            },
            Object {
              "color": "#F28D35",
            },
          ]
        }
      >
        Reserviert für 30min
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        Bitte begib dich zu dem oben genannten BIKE NEST. Du kannst dich Navigieren lassen oder es direkt aufschließen, falls du schon hier bist.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F2F2F2",
            "borderRadius": 38,
            "elevation": 3,
            "height": 55,
            "justifyContent": "center",
            "margin": 9,
            "opacity": 1,
            "padding": 18,
            "width": 328,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#F2F2F2",
                  "fontSize": 14,
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                },
                Object {
                  "color": "#000000",
                },
              ]
            }
          >
            Navigation
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F2F2F2",
            "borderRadius": 38,
            "elevation": 3,
            "height": 55,
            "justifyContent": "center",
            "margin": 9,
            "opacity": 1,
            "padding": 18,
            "width": 328,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#F2F2F2",
                  "fontSize": 14,
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                },
                Object {
                  "color": "#000000",
                },
              ]
            }
          >
            Ich bin schon hier
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Profil
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          (un)lock
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Navigation
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Verlauf
        </Text>
      </View>
    </View>
  </View>
</View>
`;
